src/process/native/win32.js

```diff
@@ -1,8 +1,88 @@
-import { exec } from 'child_process';
-
-export const getProcesses = () => new Promise(res => exec(`wmic process get ProcessID,ExecutablePath /format:csv`, (e, out) => {
-  res(out.toString().split('\r\n').slice(2).map(x => {
-    const parsed = x.trim().split(',').slice(1).reverse();
-    return [ parseInt(parsed[0]) || parsed[0], parsed[1] ];
-  }).filter(x => x[1]));
-}));
+import koffi from 'koffi';
+
+// Load Windows API
+const psapi = koffi.load('psapi.dll');
+const kernel32 = koffi.load('kernel32.dll');
+const ntdll = koffi.load('ntdll.dll');
+
+// Define Alias
+const DWORD = koffi.alias('DWORD', 'uint32_t');
+const BOOL = koffi.alias('BOOL', 'int32_t');
+const HANDLE = koffi.pointer('HANDLE', koffi.opaque())
+
+const UNICODE_STRING = koffi.struct('UNICODE_STRING', {
+  Length: 'uint16',
+  MaximumLength: 'uint16',
+  Buffer: HANDLE
+});
+
+const SYSTEM_PROCESS_ID_INFORMATION = koffi.struct('SYSTEM_PROCESS_ID_INFORMATION', {
+  ProcessId: HANDLE,
+  ImageName: UNICODE_STRING
+});
+
+const EnumProcesses = psapi.func('BOOL __stdcall EnumProcesses(_Out_ DWORD *lpidProcess, DWORD cb, _Out_ DWORD *lpcbNeeded)')
+const GetLastError = kernel32.func('DWORD GetLastError()')
+const NtQuerySystemInformation = ntdll.func('NtQuerySystemInformation', 'int32', ['int32', 'SYSTEM_PROCESS_ID_INFORMATION*', 'uint32', HANDLE]);
+
+const SystemProcessIdInformation = 88; // SYSTEM_INFORMATION_CLASS enum value for SystemProcessIdInformation
+
+const STATUS_INFO_LENGTH_MISMATCH = 0xC0000004;
+const NT_SUCCESS = (status) => status >= 0;
+const NT_ERROR = (status) => status < 0;
+
+// Using undocumented function NtQuerySystemInformation()
+// to circumvent limited privilege of wmic or gwmi
+// when querying executable path of a process
+const getProcessImageName = (pid) => {
+  let bufferSize = 1024;
+  let buffer = Buffer.alloc(bufferSize);
+
+  while (true) {
+      const info = {
+        ProcessId: pid,
+        ImageName: {
+            Length: 0,
+            MaximumLength: buffer.length,
+            Buffer: buffer
+        }
+      };
+
+      const result = NtQuerySystemInformation(SystemProcessIdInformation, info, 24, null);
+
+      if (NT_ERROR(result) && result !== STATUS_INFO_LENGTH_MISMATCH) {
+        console.error(`NtQuerySystemInformation() failed with pid = ${pid}, error = ${result}`);
+        return null;
+      }
+
+      if (NT_SUCCESS(result)) {
+        return buffer.subarray(0, buffer.length).toString('utf16le');
+      }
+
```

Comment:
```suggestion

      if (bufferSize >= 0xffff) {
        console.error(`NtQuerySystemInformation() failed with pid = ${pid}, result could not fit in buffer of size 0xffff`)
        return null;
      }

```
Just to prevent infinite loops